internal/models/oauth_state.go

```diff
@@ -0,0 +1,69 @@
+package models
+
+import (
+	"database/sql"
+	"github.com/pkg/errors"
+	"github.com/supabase/gotrue/internal/storage"
+	"time"
+
+	"github.com/gofrs/uuid"
+)
+
+type OAuthState struct {
+	ID                   uuid.UUID `json:"id" db:"id"`
+	UserID               uuid.UUID `json:"user_id" db:"user_id"`
```

Comment:
was thinking that we should tie the `OAuthState` to a session and use the `SessionId` here instead

Reply:
Interesting, we just need to be careful as the session has to be created together with a row in this table so the reference is correct.